src/ome_types/util.py

```diff
@@ -47,3 +89,166 @@ def collect_ids(value: Any) -> Dict[LSID, OMEType]:
                 ids.update(collect_ids(getattr(value, f)))
     # Do nothing for uninteresting types.
     return ids
+
+
+@lru_cache
```

Comment:
```suggestion
@lru_cache()
```

(py 3.7 tests are failing because lru_cache needed to be called back then)